lib/metrics/client-metrics.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+/**
+ * Represents a base class that is used to measure events from the server and the client as seen by the driver.
+ * @alias module:metrics~ClientMetrics
+ * @interface
+ */
+class ClientMetrics {
```

Comment:
I was thinking maybe it would be nice to provide more information as parameters (like Host where error was encountered), but then I think we would start veering into conflating with the Request Tracker implementation and I remembered your l[comment on NODEJS-300](https://datastax-oss.atlassian.net/browse/NODEJS-300?focusedCommentId=50558&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-50558) which covers this.

I think if anyone wants to do something very specific, such as Node-level metrics, they could just use `RequestTracker`, so I like this approach where the API is very simple as it makes it very easy for users to wire in their own functionality.